services/lfs/server.go

```diff
@@ -495,10 +496,27 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
 		accessMode = perm.AccessModeWrite
 	}
 
+	if ctx.Doer.IsActions() {
```

Comment:
You cannot assume `ctx.Doer` is not `nil`. I still think the previous reversion is better.